src/nodes/display/GTAONode.js

```diff
@@ -0,0 +1,364 @@
+		_quadMesh.render( renderer );
+
+		// restore
+
+		renderer.setRenderTarget( currentRenderTarget );
+		renderer.setMRT( currentMRT );
+		renderer.setClearColor( _currentClearColor, currentClearAlpha );
+		textureNode.value = currentTexture;
+
+	}
+
+	setup( builder ) {
+
+		const { textureNode, depthNode, normalNode, noiseNode } = this;
+
+		const uvNode = uv();
+
+		// const sampleTexture = ( uv ) => textureNode.uv( uv );
+		const sampleDepth = ( uv ) => depthNode.uv( uv ).x;
+		const sampleNormal = ( uv ) => normalNode.uv( uv );
+		const sampleNoise = ( uv ) => noiseNode.uv( uv );
+
+		const getSceneUvAndDepth = tslFn( ( [ sampleViewPos ] )=> {
+
+			const sampleClipPos = cameraProjectionMatrix.mul( sampleViewPos, 1.0 );
+			const sampleUv = sampleClipPos.xy.div( sampleClipPos.w.mul( 0.5 ).add( 0.5 ) );
+			const sampleSceneDepth = sampleDepth( sampleUv );
+			return vec3( sampleUv, sampleSceneDepth );
+
+		} );
+
+		const getViewPosition = tslFn( ( [ screenPosition, depth ] )=> {
+
+			const clipSpacePosition = vec4( vec3( screenPosition.xy, depth ).mul( 2.0 ).sub( 1.0 ), 1.0 );
+			const viewSpacePosition = cameraProjectionMatrixInverse.mul( clipSpacePosition );
+
+			return viewSpacePosition.xyz.div( viewSpacePosition.w );
+
+		} );
+
+		const ao = tslFn( () => {
+
+			const depth = sampleDepth( uvNode );
+
+			depth.greaterThanEqual( 1.0 ).discard();
+
+			const viewPosition = getViewPosition( uvNode, depth );
+			const viewNormal = sampleNormal( uvNode ).rgb.normalize();
+
+			const radiusToUse = this.radius;
+
+			const noiseResolution = textureSize( this.noiseNode, 0 );
+			const noiseUv = uvNode.xy.mul( this.resolution.div( noiseResolution ) );
+			const noiseTexel = sampleNoise( noiseUv );
+			const randomVec = noiseTexel.xyz.mul( 2.0 ).sub( 1.0 );
+			const tangent = vec3( randomVec.xy, 0.0 ).normalize();
+			const bitangent = vec3( tangent.y.mul( - 1.0 ), tangent.x, 0.0 );
+			const kernelMatrix = mat3( tangent, bitangent, vec3( 0.0, 0.0, 1.0 ) );
+
+			const DIRECTIONS = this.SAMPLES.lessThan( 30 ).cond( 3, 5 );
+			const STEPS = this.SAMPLES.add( DIRECTIONS.sub( 1 ) ).div( DIRECTIONS );
+
+			let ao = float( 0 );
+
+			loop( DIRECTIONS, ( { i } ) => {
+
+				const angle = float( i ).div( DIRECTIONS ).mul( PI );
+				const sampleDir = vec4( cos( angle ), sin( angle ), 0.0, noiseTexel.w.mul( 0.5 ).add( 0.5 ) );
+				sampleDir.xyz = kernelMatrix.mul( sampleDir.xyz ).normalize();
+
+				const viewDir = viewPosition.mul( - 1.0 ).normalize();
+				const sliceBitangent = sampleDir.xyz.cross( viewDir ).normalize();
+				const sliceTangent = sliceBitangent.cross( viewDir );
+				const normalInSlice = viewNormal.sub( sliceBitangent.mul( viewNormal.dot( sliceBitangent ) ) );
+
+				const tangentToNormalInSlice = normalInSlice.cross( sliceBitangent );
+				const cosHorizons = vec2( viewDir.dot( tangentToNormalInSlice ), viewDir.dot( tangentToNormalInSlice.mul( - 1.0 ) ) );
+
+				loop( STEPS, ( { i } ) => {
```

Comment:
Maybe it would be:
```js
loop( { end: STEPS, name: 'j' }, ( { j } ) => {
```

I will replace `end` for `count` for this specific approach soon.

Reply:
`j` was undefined in this case hence I've used `i`. 

There are a couple of other issues I need to investigate step by step. This is one that I wanted to report here soon^^.

Reply:
Okay, the usage of `name` fix it 👍 .